osr/interfaces/bankscreen.simba

```diff
@@ -651,7 +667,101 @@ begin
   end;
 end;
 
+function TRSPinScreen._Find(): Boolean;
+var
+  P: TPoint;
+begin
+  Result := srl.FindText(p, 'Bank of RuneScape', 'UpChars07_s', 255, 0, Box(21,30,151,50));
+end;
+
+function TRSPinScreen.IsOpen(WaitTime: UInt32 = 0): Boolean;
+var
+  t: UInt64;
+begin
+  t := GetTickCount64() + WaitTime;
+
+  repeat
+    if (Self._Find()) then
+      Exit(True)
+    else if (WaitTime > 0) then
+      Wait(Random(25, 50));
+  until (GetTickCount64() >= t);
+end;
+
+function TRSPinScreen.EnterPin(): Boolean;
+var
+  I, J, F, Start: Int32;
+  TSA: TStringArray = ['First', 'Now', 'Time', 'Finally'];
+  T: UInt64;
+  P: TPoint;
+begin
+  T := GetTickCount64 + RandomRange(4000, 8000);
+  while (CountColor(32767, 37, 107, 372, 318) <> 322) do
+  begin
+    Wait(RandomRange(100, 400));
+    if (GetTickCount64 > T) then
+      Exit(False);
+  end;
+
+  for Start := 1 to 4 do
+  begin
+    if srl.FindText(p, TSA[Start - 1], 'UpChars07_s', Box(160, 80, 350, 100)) then
+      Break;
+  end;
+
+  for J := Start to 4 do
+  begin
+    P := [0, 0];
+    if srl.FindText(P, Players.GetCurrent()^.BankPin[J], 'UpChars07_s', Box(37, 107, 372, 318)) then
+    begin
+      for I := Low(FSlots) to High(FSlots) do
+      begin
+        if PointInBox(P, FSlots[I]) then
+        begin
+          Mouse.Click(FSlots[I], Mouse_Left);
+          Break;
+        end;
+      end;
+    end else
+    begin
+      GetMousePos(P.X, P.Y);
+      for I := Low(FSlots) to High(FSlots) do
+      begin
+        if PointInBox(P, FSlots[I]) then
+        begin
+          Mouse.Click(FSlots[I], Mouse_Left);
+          Break;
+        end;
+      end;
+    end;
+
+    T := GetTickCount64 + RandomRange(4000, 8000);
+    while srl.FindText(p, TSA[J - 1], 'UpChars07_s', Box(160, 80, 350, 100)) and (CountColor(32767, 37, 107, 372, 318) <> 332) do
+    begin
+      Wait(RandomRange(300, 600));
+      if (GetTickCount64 > T) then
+        Exit(False);
+    end;
+    Wait(RandomRange(600, 1200));
+  end;
+
+  T := GetTickCount64 + RandomRange(15000, 20000);
+  while not BankScreen.IsOpen() do
+  begin
+    if ChatBox.ClickContinue then
+    begin
+      Inc(F);
+      if (F > 2) then
+        srl.Writeln('Your pin is wrong!', dtFatal);
+      Self.EnterPin;
+    end;
+    if (GetTickCount64 > T) then
+      Exit(False);
+  end;
+  Result := Self.IsOpen(RandomRange(2000, 6000));
```

Comment:
Shouldn't this be is not open? If I enter the pin correctly the bankscreen should be open, not the pinscreen(which is self.isopen)